<a href="https://colab.research.google.com/github/sauravpanchal/ai-ml-for-coders-by-laurence-moroney/blob/main/tf1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# References & Aknowledgements :

Book : AI & Machine Learning For Coders

Author : Laurence Moroney

E-book : [View here !](https://drive.google.com/file/d/13I47jJ1P0TibmrcwHEpSi2fIb-DhQpyE/view?usp=sharing)

In [1]:
import tensorflow as tf, numpy as np
from tensorflow.keras import Sequential
from tensorflow.keras.layers import Dense

In [6]:
l0 = Dense(units=1, input_shape=[1])
model = Sequential([l0])
model.compile(optimizer='sgd', loss='mean_squared_error')

xs = np.array([-1.0, 0.0, 1.0, 2.0, 3.0, 4.0], dtype=float)
ys = np.array([-3.0, -1.0, 1.0, 3.0, 5.0, 7.0], dtype=float)

model.fit(xs, ys, epochs=500)

print(model.predict([10.0]))

Epoch 1/500
1/1 [==============================] - 0s 268ms/step - loss: 0.9752
Epoch 2/500
1/1 [==============================] - 0s 9ms/step - loss: 0.8937
Epoch 3/500
1/1 [==============================] - 0s 8ms/step - loss: 0.8269
Epoch 4/500
1/1 [==============================] - 0s 7ms/step - loss: 0.7719
Epoch 5/500
1/1 [==============================] - 0s 8ms/step - loss: 0.7261
Epoch 6/500
1/1 [==============================] - 0s 5ms/step - loss: 0.6876
Epoch 7/500
1/1 [==============================] - 0s 9ms/step - loss: 0.6549
Epoch 8/500
1/1 [==============================] - 0s 5ms/step - loss: 0.6269
Epoch 9/500
1/1 [==============================] - 0s 8ms/step - loss: 0.6025
Epoch 10/500
1/1 [==============================] - 0s 6ms/step - loss: 0.5811
Epoch 11/500
1/1 [==============================] - 0s 6ms/step - loss: 0.5621
Epoch 12/500
1/1 [==============================] - 0s 7ms/step - loss: 0.5450
Epoch 13/500
1/1 [==============================] - 0s 12ms

In [7]:
print("Here is what I learned: {}".format(l0.get_weights()))

Here is what I learned: [array([[1.998069]], dtype=float32), array([-0.9940134], dtype=float32)]


From above one can determine relationship between X & Y as follows :

Y = 1.998069(X) - 0.9940134

Pretty close to actual relationship Y = 2(X) - 1